### ISSUE TITLE
Optional garbage collection of finished Workloads

### ISSUE THREAD

**Issue by woehrl01** [kubernetes-sigs/kueue#1618]
**What would you like to be added**:

I would like to have an manager option to delete workload resources as soon as (or with a ttl) the scheduled Job is finished.

**Why is this needed**:

 I changed a configuration to retain more history of job executions of a cronjob, and the memory consumption of the kueue-manager more than doubled:

![Bildschirmfoto 2024-01-19 um 11 57 50](https://github.com/kubernetes-sigs/kueue/assets/2535856/c6ff8083-f69e-4268-b24a-edf7a16f3fc7)

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

**Comment by woehrl01** [comment]
I guess the admitted workload bug this is fixed by #1654. It would be still nice to remove the workload resource all together.

**Comment by alculquicondor** [comment]
Just the Workload but keep the parent Job (or Job CRD)?

**Comment by woehrl01** [comment]
@alculquicondor yes. That was the idea. The workload will be deleted eventually, but this would free up etcd storage until ttl of the job crd has been reached.

**Comment by alculquicondor** [comment]
/retitle Optional garbage collection of finished Workloads

🤔 maybe we can also do this for orphan Workloads #1789

**Comment by kannon92** [comment]
Opening due to the work seems to be in flight.

**Comment by mimowo** [comment]
/remove-lifecycle rotten
cc @mwysokin would you like to continue working on this?
